service/src/test/java/cz/muni/fi/pa165/mushrooms/service/MushroomHunterServiceTest.java

```diff
@@ -0,0 +1,374 @@
+                    }
+                    if (mushroomHunter.getFirstName() == null){
+                        throw new IllegalArgumentException("null first name on mushroom hunter update");
+                    }
+                    if (mushroomHunter.getSurname() == null){
+                        throw new IllegalArgumentException("null surname on mushroom hunter update");
+                    }
+                    if (mushroomHunter.getUserNickname() == null){
+                        throw new IllegalArgumentException("null nickname on mushroom hunter update");
+                    }
+                    if (TestUtils.checkMushroomHunterDuplicity(persistedMushroomHunter, mushroomHunter)){
+                        throw new IllegalArgumentException("duplicating unique mushroom hunter attribute on updating");
+                    }
+                    persistedMushroomHunter.replace(mushroomHunter.getId(), mushroomHunter);
+                }
+            };
+            minTimes = 0;
+
+        }};
+
+        mushroomHunterDao.create(persistedHunter);
+        mushroomHunterDao.create(persistedHunter2);
+    }
+
+    @Test
+    public void findMushroomHunterById() throws Exception {
+        mushroomHunterDao.create(newHunter);
+        assertThat(service.findHunterById(newHunter.getId())).isEqualTo(newHunter);
+    }
+
+    @Test
+    public void findMushroomHunterByNullId() throws Exception {
+        assertThatThrownBy(() -> service.findHunterById(null)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void findMushroomHunterWithNonExistingId() throws Exception {
+        assertThat(service.findHunterById(-1L)).isNull();
+    }
+
+    @Test
+    public void findAllMushroomHunters() throws Exception {
+        mushroomHunterDao.create(newHunter);
+        assertThat(service.findAllHunters()).containsExactlyInAnyOrder(persistedHunter,persistedHunter2, newHunter);
+    }
+
+    @Test
+    public void findMushroomHunterByNickname() throws Exception {
+        String existingNickname = persistedHunter.getUserNickname();
+
+        assertThat(service.findHunterByNickname(existingNickname)).isNotNull();
+        assertThat(service.findHunterByNickname(existingNickname)).isEqualTo(persistedHunter);
+    }
+
+    @Test
+    public void findByMushroomHunterNullNickname() throws Exception {
+        assertThatThrownBy(() -> service.findHunterByNickname(null)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void findByMushroomHunterNonExistingNickname() throws Exception {
+        String nonPersistedNickname = "nickname";
+        assertThat(service.findHunterByNickname(nonPersistedNickname)).isNull();
+    }
+
+    @Test
+    public void createMushroomHunter() throws Exception {
+        service.registerHunter(newHunter, newHunter.getPasswordHash());
+        assertThat(mushroomHunterDao.findById(newHunter.getId())).isEqualTo(newHunter);
```

Comment:
dependent test - does register AND find. Assert that the collection changed (persistedMushroomHunter map)

Reply:
resolved